apps/build-onchain-apps/src/hooks/useBlockExplorerLink.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+import { Chain } from 'viem/chains';
+
+export enum HashType {
+  Address = 'address',
+  Transaction = 'tx',
+}
+/**
+ * A React hook to generate an BlockExplorer (e.g. etherscan) link.
+ * @param {Object} chain - The chain object from wagmi.
+ * @param {string} hash - The contract or transaction hash.
+ * @returns {string} - The URL to the Etherscan page for the given hash.
+ */
+function useBlockExplorerLink(chain: Chain, hash: string, type: HashType = HashType.Address) {
```

Comment:
If this is a Hook that can be reused, we should probably moved inside the `onchain` folder.